executor/insert_test.go

```diff
@@ -830,3 +830,47 @@ func (s *testSuite3) TestDMLCast(c *C) {
 	tk.MustExec("delete from t where a = ''")
 	tk.MustQuery(`select * from t`).Check(testkit.Rows())
 }
+
+func (s *testSuite3) TestAutoIDIncrementAndOffset(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec(`use test`)
+	// Test handle is PK.
+	tk.MustExec(`create table io (a int key auto_increment)`)
+	tk.Se.GetSessionVars().AutoIncrementOffset = 10
+	tk.Se.GetSessionVars().AutoIncrementIncrement = 2
+	tk.MustExec(`insert into io values (),(),()`)
+	tk.MustQuery(`select * from io`).Check(testkit.Rows("10", "12", "14"))
+	tk.MustExec(`delete from io`)
```

Comment:
It seems this behavior is incompatible with MySQL 8.0:
```sql
mysql root@172.17.0.2:test> create table t (col int key auto_increment)
Query OK, 0 rows affected
Time: 0.040s
mysql root@172.17.0.2:test> SET @@auto_increment_increment=2
Query OK, 0 rows affected
Time: 0.001s
mysql root@172.17.0.2:test> SET @@auto_increment_offset=10
Query OK, 0 rows affected
Time: 0.001s
mysql root@172.17.0.2:test> insert into t values (), (), ()
Query OK, 3 rows affected
Time: 0.016s
mysql root@172.17.0.2:test> select * from t;
+-----+
| col |
+-----+
| 2   |
| 4   |
| 6   |
+-----+
3 rows in set
Time: 0.007s
mysql root@172.17.0.2:test> 
```

As the documentation says, 
> When the value of auto_increment_offset is greater than that of auto_increment_increment, the value of auto_increment_offset is ignored.

Reply:
yes, actually is a bug itself,
https://yq.aliyun.com/articles/294121
https://jira.mariadb.org/browse/MDEV-6839
Aili and Mariadb both complain about this, Cong and I wanna fix it.

Reply:
Good catch @tangenta 

The behavior of MySQL in such cases is actually random, that is terrible. Fixing it is a better idea, and we should point this out in the official document later. @AilinKid